static/js/services/report-view-model-generator.js

```diff
@@ -0,0 +1,65 @@
+var _ = require('underscore');
+
+angular.module('inboxServices').factory('ReportViewModelGenerator',
```

Comment:
Hmm I read the google doc but I don't really get these names, I think it's because too many of these words have multiple meanings.
GetReportForViewing? GetReportForTemplate?